client/components/RateModal.jsx

```diff
@@ -33,29 +37,62 @@ export const RateModal = ({ setRate }) => {
     setOpen(false);
   };
 
+  const api = useAPI();
+  const { currentUser } = useAuth();
+
   return (
     <div>
-      <button type="button" onClick={handleOpen}>
-        react-transition-group
-      </button>
+      <Button variant="contained" color="secondary" onClick={handleOpen}>
+        Review
+      </Button>
       <Modal
-        aria-labelledby="transition-modal-title"
-        aria-describedby="transition-modal-description"
         className={classes.modal}
         open={open}
         onClose={handleClose}
         closeAfterTransition
         BackdropComponent={Backdrop}
-        BackdropProps={{
-          timeout: 500,
-        }}
+        BackdropProps={{ timeout: 500 }}
       >
         <Fade in={open}>
           <div className={classes.paper}>
-            <h2>Please rate the quiz.</h2>
-            <h4>( click on the stars below )</h4>
-            <Rate handleClose={handleClose} setRate={setRate} />
-            <p id="transition-modal-description" />
+            {/* instructions */}
+            <h2>How did you like the quiz?</h2>
+
+            {/* rate via stars */}
+            <Grid container justify="center">
+              {/* omit handleclose hook if not being used */}
+              <Rate handleClose={handleClose} setRate={setRate} />
+            </Grid>
+
+            {/* Comment Field */}
+            <br />
+            <CommentField setComment={setComment} />
+            <br />
+
+            {/* Send button */}
+            <Grid container justify="flex-end">
+              <Button
+                variant="contained"
+                color="primary"
+                onClick={() => {
+                  handleClose();
+                  if (rateVal > 0 || commentVal.length > 0) {
+                    api
+                      .post('/quizzes/rating', {
+                        User: currentUser.uid,
```

Comment:
Wait sorry, let's use the User from the API's middleware rather than passing it in

Reply:
okay! fixed! Utilizing auth middleware passed as a param into the rating route